src/components/OntologyExplorer/index.tsx

```diff
@@ -131,22 +136,65 @@ export default function OntologyExplorer({ graph }: OntologyExplorerProps): JSX.
 
     Side effect: sets the render & simulation state.
     */
+
     if (dagState) {
-      const { nodes, links } = dagState;
+      const { nodes, links } = dagState;      
+
+      const nodeToHullRoot = new Map();
+      let flag = true;
+      let height = 7;
+      const allHullRoots: string[] = [];
+      while (flag && height >= 2) {
+        flag=false;
+        const hullRoots = [...heightMap].filter(([_, v]) => v === height ).map(([k,_])=>k) // get all hullRoots of a certain height
+        hullRoots.forEach((item)=>{ // for each root
+          const hullNodes = getHullNodes(item, ontology, nodes);
+          hullNodes.forEach((n: any)=>{ // for each node in root
+            if (!nodeToHullRoot.has(n.id)) { // if not already assigned a root
+              if (!allHullRoots.includes(item)) {
+                allHullRoots.push(item); // add root
+              }
+              nodeToHullRoot.set(n.id,item); // set the root.
+            }
+          })
+        })
+        try{
+          nodes.forEach((node)=>{ // for each node,
+            const id = node.id;
+            if (!nodeToHullRoot.has(id)) { // check if node doesn't have a root
+              flag=true; // set true
+              throw(Error);
+            }
+          })
+        } catch {}
```

Comment:
generally an empty catch is an anti-pattern.  Are you just trying to terminate the loop? if so, it would be far cleaner to use a simple for-of loop and a break.